src/realm/list.hpp

```diff
@@ -785,34 +1055,24 @@ inline void Lst<T>::insert_null(size_t ndx)
 template <class T>
 inline void Lst<T>::insert_any(size_t ndx, Mixed val)
 {
-    if constexpr (std::is_same_v<T, Mixed>) {
-        insert(ndx, val);
+    if (val.is_null()) {
+        insert_null(ndx);
     }
     else {
-        if (val.is_null()) {
-            insert_null(ndx);
-        }
-        else {
-            insert(ndx, val.get<typename util::RemoveOptional<T>::type>());
-        }
+        insert(ndx, val.get<typename util::RemoveOptional<T>::type>());
     }
 }
 
 template <class T>
 size_t Lst<T>::find_any(Mixed val) const
 {
-    if constexpr (std::is_same_v<T, Mixed>) {
-        return find_first(val);
+    if (val.is_null()) {
```

Comment:
Isn't this going to change how we treat NULL values for `MIxed` types?

Reply:
But this is not for Mixed. All that was moved to a template specialization.